pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -2506,9 +2506,36 @@ func (r *ClickHouseReader) ListErrors(ctx context.Context, queryParams *model.Li
 
 	var getErrorResponses []model.Error
 
-	query := fmt.Sprintf("SELECT any(exceptionType) as exceptionType, any(exceptionMessage) as exceptionMessage, count() AS exceptionCount, min(timestamp) as firstSeen, max(timestamp) as lastSeen, any(serviceName) as serviceName, groupID FROM %s.%s WHERE timestamp >= @timestampL AND timestamp <= @timestampU GROUP BY groupID", r.TraceDB, r.errorTable)
+	query := "SELECT any(exceptionMessage) as exceptionMessage, count() AS exceptionCount, min(timestamp) as firstSeen, max(timestamp) as lastSeen, groupID"
+	if len(queryParams.ServiceName) != 0 {
+		query = query + ", serviceName"
+	} else {
+		query = query + ", any(serviceName) as serviceName"
+	}
+	if len(queryParams.ExceptionType) != 0 {
+		query = query + ", exceptionType"
+	} else {
+		query = query + ", any(exceptionType) as exceptionType"
+	}
+	query += fmt.Sprintf(" FROM %s.%s WHERE timestamp >= @timestampL AND timestamp <= @timestampU", r.TraceDB, r.errorTable)
 	args := []interface{}{clickhouse.Named("timestampL", strconv.FormatInt(queryParams.Start.UnixNano(), 10)), clickhouse.Named("timestampU", strconv.FormatInt(queryParams.End.UnixNano(), 10))}
-	if len(queryParams.OrderParam) != 0 {
+
+	if len(queryParams.ServiceName) != 0 {
+		query = query + " AND serviceName = @serviceName"
+		args = append(args, clickhouse.Named("serviceName", queryParams.ServiceName))
+	}
+	if len(queryParams.ExceptionType) != 0 {
+		query = query + " AND exceptionType = @exceptionType"
+		args = append(args, clickhouse.Named("exceptionType", queryParams.ExceptionType))
+	}
+	query = query + " GROUP BY groupID"
+	if len(queryParams.ServiceName) != 0 {
+		query = query + ", serviceName"
+	}
+	if len(queryParams.ExceptionType) != 0 {
+		query = query + ", exceptionType"
+	}
+	if len(queryParams.OrderParam) != 0 && len(queryParams.ServiceName) == 0 && len(queryParams.ExceptionType) == 0 {
```

Comment:
@makeavish isn't this incorrect? A user can still filter by `serviceName` and `ExceptionType` and order by the same.

Reply:
Yes, it should be updated.
It was added to prevent ORDER BY in case we have already filtered by exact `serviceName`